LibOS/shim/src/shim_init.c

```diff
@@ -492,14 +496,15 @@ int init_manifest (PAL_HANDLE manifest_handle)
         sys_printf("Unable to read manifest file: %s\n", errstring);
         goto fail;
     }
-    root_config = new_root_config;
 
+    root_config = new_root_config;
     return 0;
+
 fail:
-    if (bkeep_unmap_on_fail)
-        bkeep_munmap(addr, ALIGN_UP(size), VMA_INTERNAL);
-    if (stream_unmap_on_fail)
-        DkStreamUnmap(addr, size);
+    if (addr == PAL_CB(manifest_preload.start))
```

Comment:
`addr` may be uninitialized here

Reply:
It should be `!=`.

Reply:
Even after the above fixes it can call `DkStreamUnmap` on non-stream memory. Please revert those changes, the previous version was correct.